### PR TITLE
refine FlashService

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #4609 

Problem Summary:

### What is changed and how it works?
Do not pass IServer into FlashService.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by LittleFall — APPROVED** [review]

File: dbms/src/Flash/FlashService.h [hunk 1]
```diff
@@ -46,7 +46,7 @@ class FlashService : public tikvpb::Tikv::Service
     , private boost::noncopyable
 {
 public:
-    explicit FlashService(IServer & server_);
+    explicit FlashService(const TiFlashSecurityConfig & security_config_, Context & context_);
```
Comment (gengliqi) [inline dbms/src/Flash/FlashService.h:49]:
```suggestion
    FlashService(const TiFlashSecurityConfig & security_config_, Context & context_);
```

File: dbms/src/Flash/FlashService.h [hunk 2]
```diff
@@ -103,8 +103,8 @@ class AsyncFlashService final : public FlashService
     // 48 is EstablishMPPConnection API ID of GRPC
     // note: if the kvrpc protocal is updated, please keep consistent with the generated code.
     static constexpr int EstablishMPPConnectionApiID = 48;
-    explicit AsyncFlashService(IServer & server)
-        : FlashService(server)
+    explicit AsyncFlashService(const TiFlashSecurityConfig & security_config_, Context & context_)
```
Comment (gengliqi) [inline dbms/src/Flash/FlashService.h:106]:
```suggestion
    AsyncFlashService(const TiFlashSecurityConfig & security_config_, Context & context_)
```

**Review by gengliqi — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]
LGTM

File: dbms/src/Flash/FlashService.cpp [hunk 3]
```diff
@@ -44,15 +44,15 @@ extern const int NOT_IMPLEMENTED;
 
 constexpr char tls_err_msg[] = "common name check is failed";
 
-FlashService::FlashService(IServer & server_)
-    : server(server_)
-    , security_config(server_.securityConfig())
+FlashService::FlashService(const TiFlashSecurityConfig & security_config_, Context & context_)
+    : security_config(security_config_)
+    , m_context(context_)
```
Comment (fuzhe1989) [inline dbms/src/Flash/FlashService.cpp:49]: `m_context` is not recommended by ClickHouse's coding style.

**Pull request merged**